### PR TITLE
Skip schedule deleting bindings

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
we should skip schedule deleting bindings cause it makes no sense.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -334,6 +334,12 @@ func (s *Scheduler) doScheduleBinding(namespace, name string) (err error) {
 		}
 		return err
 	}
+	if !rb.DeletionTimestamp.IsZero() {
+		s.recordScheduleResultEventForResourceBinding(rb, nil, fmt.Errorf("skip schedule deleting resourceBinding: %s/%s", rb.Namespace, rb.Name))
+		klog.V(4).Info("Skip schedule deleting resourceBinding", "resourceBinding", klog.KObj(rb))
```
Comment (RainbowMango) [inline pkg/scheduler/scheduler.go:339]:
```suggestion
		klog.V(4).InfoS("Skip schedule deleting ResourceBinding", "ResourceBinding", klog.KObj(rb))
```

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -398,6 +404,12 @@ func (s *Scheduler) doScheduleClusterBinding(name string) (err error) {
 		}
 		return err
 	}
+	if !crb.DeletionTimestamp.IsZero() {
+		s.recordScheduleResultEventForClusterResourceBinding(crb, nil, fmt.Errorf("skip schedule deleting clusterResourceBinding: %s", crb.Name))
+		klog.V(4).Info("Skip schedule deleting clusterResourceBinding", "clusterResourceBinding", klog.KObj(crb))
```
Comment (RainbowMango) [inline pkg/scheduler/scheduler.go:409]:
```suggestion
		klog.V(4).InfoS("Skip schedule deleting ClusterResourceBinding", "ClusterResourceBinding", klog.KObj(crb))
```

**Comment by whitewindmills** [comment]
@RainbowMango PTAL

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -334,6 +334,12 @@ func (s *Scheduler) doScheduleBinding(namespace, name string) (err error) {
 		}
 		return err
 	}
+	if !rb.DeletionTimestamp.IsZero() {
+		s.recordScheduleResultEventForResourceBinding(rb, nil, fmt.Errorf("skip schedule deleting resourceBinding: %s/%s", rb.Namespace, rb.Name))
```
Comment (RainbowMango) [inline pkg/scheduler/scheduler.go:338]: Should we emit an event for this scenario?
I suppose, no one would expect the ResourceBing would be scheduled during deletion phase, right?
So, why do they need a warning event?

Sorry, I didn't realize it during last review.

Comment (whitewindmills) [reply pkg/scheduler/scheduler.go:338]: maybe the administrator didn't expect that his binding was accidentally deleted by others, so I think it is more reasonable to emit an event.
in addition, referring to the design of k8s in this scenario.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/schedule_one.go#L370

Comment (RainbowMango) [reply pkg/scheduler/scheduler.go:338]: Thanks, I'm sold.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**